### PR TITLE
Task js oop

### PR DESCRIPTION
**Pull request by UA-girl**
### JS OOP

[Demo](https://ua-girl.github.io/Frogger-game/) |
[Code base](https://github.com/UA-girl/Frogger-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@UA-girl for the code review purposes we need `app.js` only. Please remove all other files and also make sure that no other files except `app.js` were changed. If they were then refactor your code so that only `app.js` contains any changes that were authored by you.

File: submissions/UA-girl/task_js_oop/app.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+Player.prototype.update = function () {
+    if (this.isWin()) {
+        this.setDefaultPosition();
+        this.updateScore(5);
+        holder.innerHTML = `Cool! You won!!!
+        Your score is ${this.score}`;
+    }
+    allEnemies.forEach(enemy => enemy.checkCollisions(player));
+    allTreasures.forEach(treasure => treasure.checkCollisions(player));
```
Comment (OleksiyRudenko) [inline submissions/UA-girl/task_js_oop/app.js:111]: Here player is onbiously a variable from outside the scope of this class.
In fact you do not need and in current scope you have a simple way to refer a player not resorting to any global variables.

`allEnemies` and `allTreasures` are also global variables. Please make the `Player` class receive them, not know them.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@UA-girl a really good job!
However there is a violation of some of the OOP/SOLID principles. Please fix.

**Comment by UA-girl** [comment]
I've tried to make fixes you mentioned ) Please, check my code. Hope I've done all right ))

**Review by OleksiyRudenko — APPROVED** [review]
@UA-girl well done!

**Pull request merged**